server/api-service/lowcoder-server/src/test/java/org/lowcoder/api/application/ApplicationApiServiceTest.java

```diff
@@ -301,7 +303,12 @@ public void testCreateApplicationSuccess() {
                 .verifyComplete();
     }
 
-    @SuppressWarnings("ConstantConditions")
+    @Test
+    public void testUpdateApplicationFailedDueToLackOfDatasourcePermissions() {
```

Comment:
It's better to keep the original test and annotate it with @Disabled annotation with a brief description. Otherwise it gives false hope that the functionality works.

`@Disabled("Disabled until properly refactored")`